src/app/shared/account/provider.service.spec.ts

```diff
@@ -1,71 +1,46 @@
 
-    let mockService: MockBackend;
-    let providerService: ProviderService;
-    let broadcaster: Broadcaster;
-    let fakeAuthService: any;
+describe('Service: Provider Service', () => {
 
+    let service: ProviderService;
+    let controller: HttpTestingController;
+    let mockLogger: jasmine.SpyObj<Logger>;
     class BroadcasterTestProvider {
       static broadcaster = new Broadcaster();
     }
 
     beforeEach(() => {
-        fakeAuthService = {
-            getToken: function() {
-              return 'eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiY2xpZW50X3Nlc3Npb24iOiJURVNUU0VTU0lPTiIsInNlc3Npb25fc3RhdGUiOiJURVNUU0VTU0lPTlNUQVRFIiwiYWRtaW4iOnRydWUsImp0aSI6ImY5NWQyNmZlLWFkYzgtNDc0YS05MTk0LWRjM2E0YWFiYzUwMiIsImlhdCI6MTUxMDU3MTMxOSwiZXhwIjoxNTEwNTgwODI3fQ.l0m6EFvk5jbND3VOXL3gTkzTz0lYQtPtXS_6C24kPQk';
-            },
-            isLoggedIn: function() {
-              return true;
-            }
-          };
-
-        TestBed.configureTestingModule({
-          providers: [
-            BaseRequestOptions,
-            ProviderService,
-            MockBackend,
-            {
-              provide: Http,
-              useFactory: (backend: MockBackend,
-                           options: BaseRequestOptions) => new Http(backend, options),
-              deps: [MockBackend, BaseRequestOptions]
-            },
-            {
-              provide: AUTH_API_URL,
-              useValue: 'https://auth.fabric8.io/api/'
-            },
-            {
-                provide: AuthenticationService,
-                useValue: fakeAuthService
-            },
-            {
-              provide: Broadcaster,
-              useValue: BroadcasterTestProvider.broadcaster
-            },
-            Logger
-          ]
-        });
+      const mockAuthenticationService: jasmine.SpyObj<AuthenticationService> = createMock(AuthenticationService);
+      mockAuthenticationService.getToken.and.returnValue('eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiY2xpZW50X3Nlc3Npb24iOiJURVNUU0VTU0lPTiIsInNlc3Npb25fc3RhdGUiOiJURVNUU0VTU0lPTlNUQVRFIiwiYWRtaW4iOnRydWUsImp0aSI6ImY5NWQyNmZlLWFkYzgtNDc0YS05MTk0LWRjM2E0YWFiYzUwMiIsImlhdCI6MTUxMDU3MTMxOSwiZXhwIjoxNTEwNTgwODI3fQ.l0m6EFvk5jbND3VOXL3gTkzTz0lYQtPtXS_6C24kPQk');
+      mockLogger = jasmine.createSpyObj<Logger>('Logger', ['error']);
+
+      TestBed.configureTestingModule({
+        imports: [HttpClientTestingModule],
+        providers: [
+          { provide: Logger, useValue: mockLogger },
+          { provide: AuthenticationService, useValue: mockAuthenticationService },
+          { provide: AUTH_API_URL, useValue: 'https://auth.fabric8.io/api/' },
+          { provide: Broadcaster, useValue: BroadcasterTestProvider.broadcaster },
+          ProviderService
+        ]
       });
+      service = TestBed.get(ProviderService);
+      controller = TestBed.get(HttpTestingController);
```

Comment:
The `controller` doesn't look used anywhere. Can a `controller.expectOne('')` and `controller.verify()` be added in the tests here? If there's nowhere where this makes sense to do, then this would seem to imply that the spec doesn't need to use the HttpClientTestingModule.

Reply:
Added a few more tests for http requests.